pkgs/servers/mail/dovecot/load-extended-modules.patch

```diff
@@ -0,0 +1,12 @@
+diff --git a/src/config/config-parser.h b/src/config/config-parser.h
+index e0a0a5b..a7fd725 100644
+--- a/src/config/config-parser.h
++++ b/src/config/config-parser.h
+@@ -1,7 +1,7 @@
+ #ifndef CONFIG_PARSER_H
+ #define CONFIG_PARSER_H
+ 
+-#define CONFIG_MODULE_DIR MODULEDIR"/settings"
++#define CONFIG_MODULE_DIR "/run/current-system/sw/lib/dovecot/modules/settings"
```

Comment:
So basically, you're replacing a hardcoded path by another?

Reply:
Yes, but this setting only affects the loading of external modules. Built-in dovecot modules do not need this change.

Reply:
Mmh mmh... I hope it's for a greater good then.

Reply:
Tested it on my server, didn't notice any obvious errors.

Reply:
Fair enough, but that doesn't mean this is better than the previous iteration :D

Reply:
I also emailed the dovecot mailing list about it, maybe they can help. Then it will work to remove this patch.

Reply:
Is there a public link to the mailing list post? Perhaps it would be nice to add a comment in the nix file to revisit this from time to time?

Reply:
> Fair enough, but that doesn't mean this is better than the previous iteration :D

The patch is greatly simplified.

Reply:
> Is there a public link to the mailing list post? Perhaps it would be nice to add a comment in the nix file to revisit this from time to time?

https://dovecot.org/mailman3/archives/list/dovecot@dovecot.org/thread/P5LEA7GX463F5VHWXNAFEX5Z25BOC2UJ/

Feel free to add more information if I missed something or didn't explain it well to them.

Reply:
It's fine, as long as we can keep track of what's going on upstream around this patch and post, it's fine.